test/e2e/json-rpc/wallet_revokePermissions.spec.js

```diff
@@ -0,0 +1,58 @@
+const { strict: assert } = require('assert');
+
+const {
+  withFixtures,
+  defaultGanacheOptions,
+  unlockWallet,
+  openDapp,
+} = require('../helpers');
+const FixtureBuilder = require('../fixture-builder');
+
+describe('Revoke Dapp Permissions', function () {
+  it('should revoke dapp permissions ', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test?.fullTitle(),
+      },
+      async ({ driver }) => {
+        await unlockWallet(driver);
+
+        await openDapp(driver);
+
+        await driver.findElement({
+          text: '0x5cfe73b6021e818b776b421b1c4db2474086a7e1',
+          css: '#accounts',
+        });
+
+        // wallet_revokePermissions request
+        const revokePermissionsRequest = JSON.stringify({
+          jsonrpc: '2.0',
+          method: 'wallet_revokePermissions',
+          params: [
+            {
+              eth_accounts: {},
+            },
+          ],
+        });
+
+        const result = await driver.executeScript(
+          `return window.ethereum.request(${revokePermissionsRequest})`,
+        );
```

Comment:
This works. Though I wonder why we shouldn't add a button to the test dapp?

Reply:
Adding the button might be the better approach and could avoid having to reload the dapp in after revoking the permissions. I'll see.